src/components/chat/MessageList.js

```diff
@@ -5,8 +5,26 @@ import Message from './Message';
 
 class MessageList extends Component {
 
+  shouldComponentUpdate(nextProps, nextState) {
+    return nextProps.messages.length > 0 
+      && nextProps.messages
+        .map(m => m.id)
+        .concat(this.props.messages.map(m => m.id))
+        .reduce((acc, id) => {
+          if(!acc.indexOf(id) === -1){
+            acc.push(id)
+          }
+          return acc;
+        }, [])
+        .length !== this.props.messages;
```

Comment:
@Spetastian :+1: I see what you did here... If the number of old messages + the number of new messages !== the number of messages already shown, re-render.

Users can't delete messages right now, which simplifies things.  Later we'll remove out-of-sight messages from the DOM for performance reasons, at which point maybe we should compare the IDs of the newest incoming message and the latest newest rendered message to determine whether a re-render should occur?

Reply:
Yeah, somehow we should compare the the subset of the messages that are visible to the user with the "same" subset from the new props.

But this logic is just to determine if we should scroll down. Later we may also have a property that tells the time when there were new messages and instead we compare that prop to determine if we should scroll down. Or show a button that says "new messages, scroll down" or whatever.